### PR TITLE
Added support for custom codec when using mongo

### PR DESCRIPTION
**Pull request by Nibell**
Ready for review @torshid

### CODE REVIEW

File: src/main/java/com/turkraft/springfilter/boot/BsonFilterArgumentResolver.java [hunk 1]
```diff
@@ -11,14 +14,18 @@
...
+import org.springframework.data.mongodb.CodecRegistryProvider;
 /**
  * Resolver for {@link org.bson.conversions.Bson Bson} and {@link org.bson.Document Document}
  * annotated with {@link Filter}.
  */
 
+@Component
 public class BsonFilterArgumentResolver implements HandlerMethodArgumentResolver {
 
+  @Autowired
```
Comment (torshid) [inline src/main/java/com/turkraft/springfilter/boot/BsonFilterArgumentResolver.java:26]: Shouldn't this have `required=false`?

Comment (Nibell) [reply src/main/java/com/turkraft/springfilter/boot/BsonFilterArgumentResolver.java:26]: Yeah, you are right 😃

**Comment by torshid** [comment]
Hi @Nibell, thanks for this contribution. I was wondering if a project using JPA but not Mongo (or a project using Mongo but not JPA) would get a ClassNotFoundException, if not I can merge your PR and release a new version.

**Comment by Nibell** [comment]
Yeah, i will test it in a vanilla spring app, jpa only app and mongo only app. And see what happen.
Right now i have running the same code in a mongo jpa app 😄

**Comment by Nibell** [comment]
Works with jpa for me using simple spring-boot app.
Removed mongo repository pattern, because need to implement auto config to make it work and need to split project to multiple folder to do that 😄

**Pull request merged**